private/buf/bufgen/features.go

```diff
@@ -22,86 +22,189 @@ import (
 	"github.com/bufbuild/buf/private/bufpkg/bufconfig"
 	"github.com/bufbuild/buf/private/bufpkg/bufimage"
 	"github.com/bufbuild/buf/private/pkg/app"
+	"google.golang.org/protobuf/reflect/protoreflect"
 	"google.golang.org/protobuf/types/descriptorpb"
 	"google.golang.org/protobuf/types/pluginpb"
 )
 
 // requiredFeatures maps a feature to the set of files in an image that
 // make use of that feature.
-type requiredFeatures map[pluginpb.CodeGeneratorResponse_Feature][]string
+type requiredFeatures struct {
+	flags                  map[pluginpb.CodeGeneratorResponse_Feature][]string
+	editions               map[descriptorpb.Edition][]string
+	minEdition, maxEdition descriptorpb.Edition
+}
+
+func newRequiredFeatures() *requiredFeatures {
+	return &requiredFeatures{
+		flags:    map[pluginpb.CodeGeneratorResponse_Feature][]string{},
+		editions: map[descriptorpb.Edition][]string{},
+	}
+}
 
 type featureChecker func(options *descriptorpb.FileDescriptorProto) bool
 
 // Map of all known features to functions that can check whether a given file
 // uses said feature.
 var allFeatures = map[pluginpb.CodeGeneratorResponse_Feature]featureChecker{
```

Comment:
global vars should always go at the top, only below global consts

Also rename to `featureToFeatureChecker`

Reply:
done